### PR TITLE
Retain dimnames in array_branch and array_tree

### PR DESCRIPTION
**Pull request by flying-sheep**
By not resetting `dim(array)` without need, the return value retains its dimnames

### CODE REVIEW

**Comment by lionel-** [comment]
Could you provide a motivation for this change please? Ideally the motivation would be provided in a NEWS bullet explaining the change and would be covered by unit tests.

**Comment by flying-sheep** [comment]
Sure, done!

**Comment by flying-sheep** [comment]
OK, this looks good to go!

File: tests/testthat/setup.R [hunk 1]
```diff
@@ -0,0 +1 @@
+Sys.setlocale("LC_MESSAGES", "C")
```
Comment (lionel-) [inline tests/testthat/setup.R]: Why did you add this setup file?

File: R/arrays.R [hunk 2]
```diff
@@ -50,6 +50,8 @@ array_branch <- function(array, margin = NULL) {
 
   if (length(margin) == 0) {
     list(array)
+  } else if (identical(margin, 1) && is.null(dim(array))) {
```
Comment (lionel-) [inline R/arrays.R]: The failure this is fixing was from an example, could you capture it in the unit tests as well please?

File: tests/testthat/setup.R (hunk 1 above)
Comment (flying-sheep) [reply tests/testthat/setup.R]: Because otherwise the tests don’t run on my localized system (the german error messages don’t match the expected english ones)

Technically that change should go into another PR, but that’s a formality, so I thought I’ll only bother if you insist.

File: R/arrays.R (hunk 2 above)
Comment (flying-sheep) [reply R/arrays.R]: Will do!

File: tests/testthat/setup.R (hunk 1 above)
Comment (lionel-) [reply tests/testthat/setup.R]: Interesting, I wonder if this should be set by testthat

File: R/arrays.R (hunk 2 above)
Comment (lionel-) [reply R/arrays.R]: Do you think the branch should be `if (is.null(dim(array)))`, and then we'd check for the margin and give an informative error message if it's not `1`?

By the way, `identical(1, 1L)` is `FALSE`. I think we need `is_integerish(margin, n = 1) && margin == 1`. Could you capture the integer case in the tests as well please?

**Comment by lionel-** [comment]
Thanks!
